.circleci/config.yml

```diff
@@ -337,3 +344,39 @@ workflows:
       - test_e2e_website:
           requires:
             - checkout
+
+  react-next:
+    # triggers:
+    #   - schedule:
+    #       cron: '0 0 * * *'
+    #       filters:
+    #         branches:
+    #           only:
+    #             - master
+    jobs:
+      - hold:
+          type: approval
```

Comment:
The downside is that all the PRs will have a pending check if this workflow is not triggered manually.

An alternative approach that avoids the pending check is to manually trigger the workflow:

https://github.com/mui/mui-x/assets/13808724/1058a213-ece4-497e-a853-4152d099d824

Thoughts?

cc @mui/code-infra

Reply:
I guess you could set up a [nightly build](https://support.circleci.com/hc/en-us/articles/115015983008-How-to-Schedule-a-Nightly-Build), but then I guess we have the problem of not being notified. I'm not sure how hard it would be to set up slack or email notifications for a single failed workflow.

Reply:
Yes, a scheduled workflow is good. But I also wanted to add an ability to run the workflow on the PR on demand.
This would be useful when working on React 19 compatibility, or when we want to make sure the specific PR would work across different React versions.

Reply:
Yep, I understand. Having those builds in pending state won't be the greatest DX though.

How hard would it be to do this workflow in GitHub actions? It would open up the options to build based on e.g. a PR comment, or PR title, or a label.

Reply:
Actually, triggering a CircleCI workflow from GitHub action if the PR has a specific label shouldn't be that hard to do 🤔
I've never used GitHub Actions, but I can give it a try, thanks for the suggestion 👍🏻

Reply:
> Having those builds in pending state won't be the greatest DX though.

Yes, it's a no-go. CircleCI is not great when it comes to optional manual workflows

Reply:
> Actually, triggering a CircleCI workflow from GitHub action if the PR has a specific label shouldn't be that hard to do

I don't think we get this info in the the pipeline, but I only investigated it briefly.

Reply:
I've reverted the approval type for this workflow.
It is still possible to run react-next tests on demand, it's just not as convenient. See https://github.com/cherniavskii/mui-x/blob/react-19-in-CI/test/README.md#next-version
The upside is that the always pending workflow is gone 👍🏻